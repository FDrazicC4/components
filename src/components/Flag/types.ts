import { StyledParentComponent, Colors as GenericColors } from '@components/types';


export type Colors = GenericColors;


export interface Props extends StyledParentComponent {

    /**
     * The color of the label.
     * @default 'info'
     */
    color?: Colors,

}
