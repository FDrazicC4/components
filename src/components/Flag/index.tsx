import React from 'react';
import clsx from 'clsx';
import { Chip } from '@material-ui/core';
import Typography from '@components/Typography';
import { Props } from './types';


const classes = {
    label: 'brew-Flag__label',
};


/**
 * The `<Flag />` can be used to indicate a specific note of data or status on data associations for a specific entity.
 */
export const Flag = React.forwardRef(({
    color = 'info',
    children,
    className,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const classNames = clsx(
        'brew-Flag',
        `brew-Flag--color-${color}`,
        { 'brew-Flag--isEmpty': !children },
        className,
    );


    return (
        <Chip
            classes={classes}
            className={classNames}
            label={(
                <Typography align="center" className="brew-Flag__content" variant="caption">
                    {children}
                </Typography>
            )}
            ref={ref}
            {...otherProps}
        />
    );


});


export default Flag;
