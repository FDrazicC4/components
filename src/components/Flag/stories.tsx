import React, { ReactElement, ReactNode } from 'react';
import { select, text } from '@storybook/addon-knobs';
import Flag from './index';
import { Colors } from './types';


export default {
    component: Flag,
    parameters: {
        componentSubtitle: <Flag color="success">Stable</Flag>,
    },
    title: 'Stable|Data Display/Flag',
};


export const General = (): ReactElement => {
    const color: Colors = select('color', ['success', 'danger', 'warning', 'info', 'light', 'dark'], 'info');
    const children: ReactNode = text('children', 'Flag');

    return (
        <Flag color={color}>{children}</Flag>
    );
};


export const Default = (): ReactElement => (
    <React.Fragment>
        <Flag>Default</Flag>
        <Flag color="success">Default</Flag>
        <Flag color="warning">Default</Flag>
        <Flag color="danger">Default</Flag>
        <Flag color="light">Default</Flag>
        <Flag color="dark">Default</Flag>
    </React.Fragment>
);


export const Empty = (): ReactElement => (
    <React.Fragment>
        <Flag />
        <Flag color="success" />
        <Flag color="warning" />
        <Flag color="danger" />
        <Flag color="light" />
        <Flag color="dark" />
    </React.Fragment>
);
