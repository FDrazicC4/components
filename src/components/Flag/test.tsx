import React from 'react';
import renderer from 'react-test-renderer';
import Flag from '@components/Flag';


it('renders correctly', () => {
    const tree = renderer
        .create((
            <React.Fragment>
                <Flag>Default</Flag>
                <Flag />
            </React.Fragment>
        ))
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('renders correctly with various props', () => {
    const tree = renderer
        .create((
            <React.Fragment>
                <Flag>Default</Flag>
                <Flag color="success">Default</Flag>
                <Flag color="warning">Default</Flag>
                <Flag color="danger">Default</Flag>
                <Flag color="light">Default</Flag>
                <Flag color="dark">Default</Flag>
            </React.Fragment>
        ))
        .toJSON();
    expect(tree).toMatchSnapshot();
});
