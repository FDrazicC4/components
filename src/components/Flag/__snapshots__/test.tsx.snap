// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="MuiChip-root brew-Flag brew-Flag--color-info"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label brew-Flag__label"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-caption brew-Typography--align-center brew-Flag__content MuiTypography-caption MuiTypography-alignCenter"
      >
        Default
      </span>
    </span>
  </div>,
  <div
    className="MuiChip-root brew-Flag brew-Flag--color-info brew-Flag--isEmpty"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label brew-Flag__label"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-caption brew-Typography--align-center brew-Flag__content MuiTypography-caption MuiTypography-alignCenter"
      />
    </span>
  </div>,
]
`;

exports[`renders correctly with various props 1`] = `
Array [
  <div
    className="MuiChip-root brew-Flag brew-Flag--color-info"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label brew-Flag__label"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-caption brew-Typography--align-center brew-Flag__content MuiTypography-caption MuiTypography-alignCenter"
      >
        Default
      </span>
    </span>
  </div>,
  <div
    className="MuiChip-root brew-Flag brew-Flag--color-success"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label brew-Flag__label"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-caption brew-Typography--align-center brew-Flag__content MuiTypography-caption MuiTypography-alignCenter"
      >
        Default
      </span>
    </span>
  </div>,
  <div
    className="MuiChip-root brew-Flag brew-Flag--color-warning"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label brew-Flag__label"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-caption brew-Typography--align-center brew-Flag__content MuiTypography-caption MuiTypography-alignCenter"
      >
        Default
      </span>
    </span>
  </div>,
  <div
    className="MuiChip-root brew-Flag brew-Flag--color-danger"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label brew-Flag__label"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-caption brew-Typography--align-center brew-Flag__content MuiTypography-caption MuiTypography-alignCenter"
      >
        Default
      </span>
    </span>
  </div>,
  <div
    className="MuiChip-root brew-Flag brew-Flag--color-light"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label brew-Flag__label"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-caption brew-Typography--align-center brew-Flag__content MuiTypography-caption MuiTypography-alignCenter"
      >
        Default
      </span>
    </span>
  </div>,
  <div
    className="MuiChip-root brew-Flag brew-Flag--color-dark"
    onKeyUp={[Function]}
  >
    <span
      className="MuiChip-label brew-Flag__label"
    >
      <span
        className="MuiTypography-root brew-Typography brew-Typography--display-initial brew-Typography--variant-caption brew-Typography--align-center brew-Flag__content MuiTypography-caption MuiTypography-alignCenter"
      >
        Default
      </span>
    </span>
  </div>,
]
`;
